conans/tools.py

```diff
@@ -183,15 +183,27 @@ def vcvars_command(settings):
                                  % (existing_version, compiler_version))
     else:
         env_var = "vs%s0comntools" % compiler_version
+
+        vs_path = None
+        if env_var == 'vs150comntools' and not env_var in os.environ:
+            try:
```

Comment:
well, probably I can write myself, if you can tell me how to cache exactly :). is there any existing python or conan facility to cache already available for use? or just cache to file/registry/whatever else?
also there are concerns about user reinstalled/upgraded VS (e.g. uninstalled Community and Installed professional instead), so when and how to invalidate cache?